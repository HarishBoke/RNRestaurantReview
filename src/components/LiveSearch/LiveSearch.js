import React, {useState} from 'react';
import {TextInput} from 'react-native';
import liveSearchStyle from './LiveSearch.style';

const LiveSearch = (props) => {
  const {setRestaurantSearch} = props;
  const [searchResult, setSearchResult] = useState(null);
  return (
    <TextInput
      style={liveSearchStyle.liveSearchInput}
      placeholder="Find Restaurant"
      onChangeText={(search) => {
        setRestaurantSearch(search);
        setSearchResult(search);
      }}
      value={searchResult}
    />
  );
};

export default LiveSearch;
