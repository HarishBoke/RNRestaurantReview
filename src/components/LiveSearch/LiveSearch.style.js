export default {
  liveSearchInput: {
    fontSize: 18,
    padding: 10,
    paddingHorizontal: 20,
    borderWidth: 1,
    borderColor: '#ddd',
    backgroundColor: '#fefefe',
  },
};
