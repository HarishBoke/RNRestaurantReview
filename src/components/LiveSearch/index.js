import LiveSearch from './LiveSearch';

export default LiveSearch;
