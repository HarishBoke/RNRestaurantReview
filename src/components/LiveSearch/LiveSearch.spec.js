import React from 'react';
import {shallow} from 'enzyme';

import LiveSearch from './LiveSearch';

const initApp = () => {
  const wrapper = shallow(<LiveSearch />);
  return {wrapper};
};

describe('Component: LiveSearch', () => {
  describe('rendering LiveSearch', () => {
    it('Snapshot: should display the LiveSearch', async () => {
      const {wrapper} = initApp();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
