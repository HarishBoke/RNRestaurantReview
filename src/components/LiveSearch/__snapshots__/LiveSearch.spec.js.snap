// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LiveSearch rendering LiveSearch Snapshot: should display the LiveSearch 1`] = `ShallowWrapper {}`;
